innerprod/fullysec/damgard_dec_multi.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2018 XLAB d.o.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fullysec
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+
+	"github.com/fentec-project/gofe/data"
+	"github.com/fentec-project/gofe/sample"
+)
+
+// DamgardMultiClient represents a client in a decentralized
+// multi client variant of the underlying multi client (dmagard_multi)
+// scheme based. The decentralization is based on
+// Abdalla, Benhamouda, Kohlweiss,and Waldner:
+// "Decentralizing Inner-Product Functional Encryption".
+// The participants in the scheme are clients without a central authority.
+// They interactively generate private keys for each client so that client i
+// can encrypt vector x_i. The scheme allows the clients to interactively
+// generate a key_Y, depending on a matrix Y with rows y_i, so that
+// given key_y and the ciphertext the decryptor can compute value
+// Σ_i <x_i, y_i> (sum of dot products).
+type DamgardDecMultiClient struct {
+	// number of encryptors
+	Idx           int
+	DamgardScheme *DamgardMulti
```

Comment:
Would it make sense (I'm not really sure on this) to embed this struct directly? This would solve the issue of having to type `c.DamgardScheme. ...` everywhere - but I'm not certain if it makes sense otherwise. What do you think?

Reply:
Not sure. `c.DamgardScheme` is also used for to call function `c.DamgardScheme.Decrypt`. Could we do something similar if it was embedded directly?

Reply:
Look at `ddh_multi.go`. I think it could be done in the same way here as it is there.